agent/xdsv2/listener_resources.go

```diff
@@ -540,33 +547,40 @@ func (pr *ProxyResources) makeEnvoyTLSParameters(defaultParams *pbproxystate.TLS
 }
 
 func (pr *ProxyResources) makeEnvoyTransportSocket(ts *pbproxystate.TransportSocket) (*envoy_core_v3.TransportSocket, error) {
+	fmt.Println("************* transport socket: ", ts)
+	//fmt.Println("************* transport socket type: ", ts.ConnectionTls.(type))
 	// TODO(JM): did this just make tests pass.  Figure out whether proxyState.Tls will always be available.
-	if pr.proxyState.Tls == nil {
-		return nil, nil
-	}
+	//if pr.proxyState.Tls == nil {
+	//	return nil, nil
+	//}
 	if ts == nil {
 		return nil, nil
 	}
 	commonTLSContext := &envoy_tls_v3.CommonTlsContext{}
+	if ts.AlpnProtocols != nil {
+		commonTLSContext.AlpnProtocols = ts.AlpnProtocols
+	}
 
 	// Create connection TLS. Listeners should only look at inbound TLS.
 	switch ts.ConnectionTls.(type) {
 	case *pbproxystate.TransportSocket_InboundMesh:
+		fmt.Println("******** inbound mesh")
 		downstreamContext := &envoy_tls_v3.DownstreamTlsContext{}
 		downstreamContext.CommonTlsContext = commonTLSContext
 		// Set TLS Parameters.
-		tlsParams := pr.makeEnvoyTLSParameters(pr.proxyState.Tls.InboundTlsParameters, ts.TlsParameters)
-		commonTLSContext.TlsParams = tlsParams
+		//tlsParams := pr.makeEnvoyTLSParameters(pr.proxyState.Tls.InboundTlsParameters, ts.TlsParameters)
```

Comment:
in both cases here and below, we could only call `makeEnvoyTLSParameters()` when `pr.poxyState.Tls` and it seems like our explicit verification would work and proxyStateConverter code that was added would both pass.  However this check and comment below require some thought:
```
		if pr.proxyState.Tls == nil {
			// if tls is nil but connection tls is provided, then the proxy state is misconfigured
			return nil, fmt.Errorf("proxyState.Tls is required to generate router's transport socket")
		}
```

Reply:
might be worth backlogging a task to investigate this further, but try and take the code changes that gets all tests and verification working at the moment.